### DISCUSSION TITLE
backfill

### DISCUSSION THREAD

**Discussion by bangbangDong** [apache/airflow#42852] [General]
### Apache Airflow version

2.10.2

### What happened?

![image](https://github.com/user-attachments/assets/a8db096b-7510-4867-a261-231302419f55)
In normal run, this would be considered normal or skipped tasks, but in backfill it would be considered a failure. (I can't keep using the --continue-on-failures argument.)

### How to reproduce

Run the history task with backfill and build the task with task.virtualenv, venv_cache_path

### Operating System

ubuntu

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment]
I am not sure what is the ask. Look a bit like you missed that you should spend a bit more time to explain what you think is wrong and what you think should happend @bangbangDong. I converted it into a discussion, but if you think that you would like to have some action as result of you report, spending a bit more of time to explain it would be a good idea, so I encourage you to do so.

**Reply by bangbangDong** [reply]
In normal operation, the example state shown in the figure above should end up being either a skip or a success state, but when using backfill to supplement historical data, this state is considered a failed state and backfill is forcibly stopped.

**Reply by bangbangDong** [reply]
![image](https://github.com/user-attachments/assets/286706e0-c725-428b-b187-992a910d11e2)
I just re-executed backfill, and all of my tasks look like this
@task
def my_task():
     pass

**Reply by bangbangDong** [reply]
![image](https://github.com/user-attachments/assets/802438db-73e1-4790-b905-a1610d88a178)
The task should be the same as the previous one, but it is marked as success to enable the task to run properly.

**Reply by bangbangDong** [reply]
In addition, another problem occurred when I was trying to backfill. In the process of backfill, all tasks in the next task have finished running, but its state is always running, which makes it impossible for backfill to continue to run. Moreover, this state seems to never stop. Sometimes it ran for days until I got him deleted
![image](https://github.com/user-attachments/assets/e6704c90-094f-4140-8ce0-e1c0f3271cac)
The above example is normal because I set --continue-on-failures in backfill.
![image](https://github.com/user-attachments/assets/3bec44b4-0fa1-4242-8b2c-899549b53f58)
However, in the same case, this task will remain in the running state and never stop. Even if all its tasks are displayed as success, the dag will still remain in the running state. Even if I set dagrun_timeout, this will not affect the situation.

**Reply by bangbangDong** [reply]
ps: My dag is set to max_active_runs=1

**Reply by potiuk** [reply]
No idea - but maybe someone will take a look at your report and will be able to understand it (there are some inconsistencies  - in one place you mention python task.virtualenv with cache , in another you say you are using "@task" and you use wording that makes it hard to understand "task never stop"?. 

Also it misses information and command you used from backfill and logs from backfill - because there likely you would see some errors (I am wildly guessing now).  

Likely providing information like pasting the DAG you use, logs from the backfill command you use might help somoene to understand and replicate what you try to do (I am not expert in this part, so I can only advice what more information you can providde and request some consistency - hopefully if you provide those, maybe someone will be able to help.
